## FALL2023 AIN212 Course Project: Free Palestine
#### Talha Kaba - 2210765037
#### Abdulkadir Parlak - 2210765025
### Dataset: Fatalities in the Israeli-Palestinian Conflict
* source: https://www.kaggle.com/datasets/willianoliveiragibin/fatalities-in-the-israeli-palestinian

In [21]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


# Step 1: Data Collection

In [22]:
df = pd.read_csv('fatalities_isr_pse_conflict_2000_to_2023.csv')

"""
filtered_df = df[df['age'] <= 18]
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)
filtered_df
"""

print("Number of rows:", len(df))
print("Number of columns:", len(df.columns))
df.head(10)

Number of rows: 11124
Number of columns: 16


name date_of_event   age  \
0  'Abd a-Rahman Suleiman Muhammad Abu Daghash    2023-09-24  32.0   
1         Usayed Farhan Muhammad 'Ali Abu 'Ali    2023-09-24  21.0   
2             'Abdallah 'Imad Sa'ed Abu Hassan    2023-09-22  16.0   
3             Durgham Muhammad Yihya al-Akhras    2023-09-20  19.0   
4                 Raafat 'Omar Ahmad Khamaisah    2023-09-19  15.0   
5                        'Ata Yasser 'Ata Musa    2023-09-19  29.0   
6                     Yusef Salem Yusef Radwan    2023-09-19  24.0   
7                Mahmoud Khaled S'ud 'Ar'arawi    2023-09-19  25.0   
8                  Mahmoud 'Ali Nafe'a a-S'adi    2023-09-19  23.0   
9                    Milad Munzer Wajih a-Ra'i    2023-09-09  15.0   

   citizenship     event_location event_location_district  \
0  Palestinian     Nur Shams R.C.                 Tulkarm   
1  Palestinian     Nur Shams R.C.                 Tulkarm   
2  Palestinian           Kfar Dan                   Jenin   
3  Palestinian  'Aqbat Jaber R.C.                 Jericho   
4  Palestinian         Jenin R.C.                   Jenin   
5  Palestinian         Jenin R.C.                   Jenin   
6  Palestinian          Gaza City                    Gaza   
7  Palestinian         Jenin R.C.                   Jenin   
8  Palestinian         Jenin R.C.                   Jenin   
9  Palestinian     al-'Arrub R.C.                  Hebron   

  event_location_region date_of_death gender took_part_in_the_hostilities  \
0             West Bank    2023-09-24      M                          NaN   
1             West Bank    2023-09-24      M                          NaN   
2             West Bank    2023-09-22      M                          NaN   
3             West Bank    2023-09-20      M                          NaN   
4             West Bank    2023-09-19      M                          NaN   
5             West Bank    2023-09-20      M                          NaN   
6            Gaza Strip    2023-09-19      M                           No   
7             West Bank    2023-09-19      M                          NaN   
8             West Bank    2023-09-19      M                          NaN   
9             West Bank    2023-09-09      M                          NaN   

  place_of_residence place_of_residence_district type_of_injury  \
0     Nur Shams R.C.                     Tulkarm        gunfire   
1     Nur Shams R.C.                     Tulkarm        gunfire   
2           al-Yamun                       Jenin        gunfire   
3  'Aqbat Jaber R.C.                     Jericho        gunfire   
4              Jenin                       Jenin        gunfire   
5              Jenin                       Jenin        gunfire   
6         Khan Yunis                  Khan Yunis        gunfire   
7         Jenin R.C.                       Jenin        gunfire   
8         Jenin R.C.                       Jenin        gunfire   
9     al-'Arrub Camp                      Hebron        gunfire   

        ammunition                killed_by  \
0  live ammunition  Israeli security forces   
1  live ammunition  Israeli security forces   
2  live ammunition  Israeli security forces   
3  live ammunition  Israeli security forces   
4  live ammunition  Israeli security forces   
5          missile  Israeli security forces   
6  live ammunition  Israeli security forces   
7          missile  Israeli security forces   
8          missile  Israeli security forces   
9  live ammunition  Israeli security forces   

                                               notes  
0  Fatally shot by Israeli forces while standing ...  
1  Fatally shot by Israeli forces while trying to...  
2  Fatally shot by soldiers while firing at them ...  
3  Shot in the head by Israeli forces while throw...  
4  Wounded by soldiers’ gunfire after running awa...  
5  Wounded by shrapnel from a missile fired by Is...  
6  Fatally shot by soldiers while participating i...  
7  Wounded by shrapnel from a missile fired by Is...  


# Attributes of the Dataset:
**Here are the descriptions of the attributes from the dataset:**
* **name:** name of the people died
* **date_of_event:** ...
.
.
.

# Step 2: Data Preprocessing and Cleaning

In [23]:
df.isnull().sum()

name                               0
date_of_event                      0
age                              129
citizenship                        0
event_location                     0
event_location_district            0
event_location_region              0
date_of_death                      0
gender                            20
took_part_in_the_hostilities    1430
place_of_residence                68
place_of_residence_district       68
type_of_injury                   291
ammunition                      5253
killed_by                          0
notes                            280
dtype: int64

There are ...
* 1430 missing values in the **took_part_in_the_hostilities** attribute.
* 129 missing values in the **age** attribute.
* 20 missing values in the **gender** attribute.
* 68 missing values in the **place_of_residence** attribute.
* 68 missing values in the **place_of_residence_district** attribute.
* 291 missing values in the **type_of_injury** attribute.
* 5253 missing values in the **ammunition** attribute.
* 280 missing values int he **notes** attribute.

In [23]:
# how to handle outliers? drop or imputation?
# are there any unnecessary attributes?
# convert categorical columns to numeric values(onehot encoding? ordinal encoding?)
# normalization of data